### ISSUE TITLE
blockhash is based on non standard block structure

### ISSUE THREAD

**Issue by sirpy** [celo-org/celo-blockchain#1981]
The vast majority of EVM based blockchains leave the standard set of fields which are used to calculate the blockhash
in recent celo updates some fields like difficulty have been removed. This causes for example, the blocknumber to be at slot 7 of the rlp header array vs slot 9 in most other EVMs. This can and will cause issues for any code trying to parse rlp headers expecting the block fields to be at the standard place.

**Comment by sirpy** [comment]
@hbandura Can you also take a look at this issue?

**Comment by sirpy** [comment]
@hbandura

**Comment by hbandura** [comment]
Can you provide examples of which tools fail due to this difference?

**Comment by sirpy** [comment]
@hbandura i dont think this is relevant, any code that parses blocks rlp headers will fail as the blocknumber is not in the same array slot as on every other evm block rlp header.

Specifically take bridges that operate based on blockheader proofs. they need to verify that the rlp header hash matches the blockhash and they need to parse the rlp header to extract the block data.

**Comment by hbandura** [comment]
To understand better the problem, please let me know if you are referring to either:
1) rlp data from rpc is not enough to correctly calculate and verify block's hash, or
2) the celo block is not 100% as an ethereum block


if it's (1) we need to address that immediately. 
If it's (2), well, it's a different blockchain, with a (marginally) different block format. If some specific libraries are needed then we may find a workaround for it.

**Comment by sirpy** [comment]
@hbandura 
its closer to 2. BUT
1. it had similar field ordering to ethereum before some chain update, so it is a BREAKING CHANGE, that probably wasn't intended.
2. all other evms i've tested have the same block header field order and if some field is unused they just keep it as a place holder with 0/null value, any new fields are added at the end.

**Comment by sirpy** [comment]
@hbandura

**Comment by sirpy** [comment]
?

**Comment by hbandura** [comment]
Regarding 1, have you identified where the breaking change ocurred for you? So I have somewhere to start investigating from

**Comment by sirpy** [comment]
@hbandura 
So actually I'm no longer sure that it is something that changed, could be that this was always the block data order on Celo.
